webapp/src/DOMJudgeBundle/Controller/AnalysisController.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+namespace DOMJudgeBundle\Controller;
+
+use DOMJudgeBundle\Service\DOMJudgeService;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use DOMJudgeBundle\Entity\Problem;
+use DOMJudgeBundle\Entity\Team;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+
+/**
+ * @Route("/jury/analysis")
+ * @Security("has_role('ROLE_JURY')")
+ */
+class AnalysisController extends Controller
+{
+    /**
+     * @var DOMJudgeService
+     */
+    private $DOMJudgeService;
+
+    private static function set_or_increment(array &$array, string $index) {
+      if (!array_key_exists($index, $array)) {
+        $array[$index] = 0;
+      }
+      $array[$index] = $array[$index] +1;
+    }
+
+    public function __construct(DOMJudgeService $DOMJudgeService)
+    {
+        $this->DOMJudgeService = $DOMJudgeService;
+    }
+
+    /**
+     *@Route("/", name="analysis_index")
+     */
+    public function indexAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $contest = $this->DOMJudgeService->getCurrentContest();
+
+        // First collect information about problems in this contest
+        $problems = $contest->getProblems(); // TODO: this gets ContestProblem, but we need Problem as well
+
+        // Need to query directly the count, otherwise symfony memory explodes
```

Comment:
I think we can also solve this by marking the testcases as extra lazy, then we can do testcase count.
I was also thinking about maybe linking Problem two two testcase entities: a `SlimTestcase` and a normal `Testcasez, where the first one doesn't contain the big fields. We often don't need them anyway. I'm not sure if we can do this with an `extend` from one to the other, but that can be tested